compat/mingw.c

```diff
@@ -1311,13 +1311,42 @@ static char *path_lookup(const char *cmd, int exe_only)
 }
 
 #if defined(_MSC_VER)
+
+/* We need a stable sort */
+#ifndef INTERNAL_QSORT
+#include "qsort.c"
+#endif
+
+/* Compare only keys */
```

Comment:
the previous version used case-insensitive routines.  the new version is doing exact match.  do we care?

Reply:
Hrm. You're right...

Reply:
... and there seems to not be any way to compare `WCHAR`s case-insensitively. I guess there are issues with compound characters or some such.

So how about this instead?

```c
size_t p_len, q_len;
int ret;

/* Find end of keys */
for (p_len = 0; p[p_len] && p[p_len] != L'='; p_len++)
    ; /* do nothing */
for (q_len = 0; q[q_len] && q[q_len] != L'='; q_len++)
    ; /* do nothing */

/* Are keys identical (modulo case)? */
if (p_len == q_len && !_wcsnicmp(p, q, p_len))
    return 0;

ret = _wcsnicmp(p, q, p_len < q_len ? p_len : q_len);
return ret ? ret : (p_len < q_len ? -1 : +1);
```

Reply:
I can't tell in this GH diff viewer, so I'll ask it here.  The deltaenv[] contains "KEY\\0" (and possibly "KEY=\\0") for items to be deleted from the existing settings.  Have we already converted them to canonical form L"KEY=\\0" by the time we get here?

Reply:
@dscho can [CompareStringEx](https://msdn.microsoft.com/en-us/library/windows/desktop/dd317761(v=vs.85).aspx) get mapped into GfW? It's kind of like `_wcsnicmp` but has extended option which might allow you to get exactly what you need.

It's Win32 API so it lacks conformance to the style of GfW, but the functionality win might be worth it.

Reply:
Sorry, I missed this before merging the PR... I had a quick look and was *overwhelmed* with the amount of options to compare case-insensitively. I never knew how many options there are!

However, I have the impression that Win32 API's environment handling considers environment keys as identical if `_wcsicmp()` considers them identical. Am I correct?